syntax = "proto3";

package gemm;


message MatMulInitRequest {
	int32 m = 1;
	int32 k = 2;
	int32 n = 3;
	string pref = 4;
}

message MatMulInitResponse {
	int32 operationId = 1;
	int32 estimatedTime = 2;
}

message MatrixChunk {
	int32 rows = 1;
	int32 cols = 2;
	int32 rowStart = 3;
	int32 colStart = 4;
	int32 operationId = 6;
	bool isLeftInputMatrix = 7;
	bool isOutputMatrix = 8;
	repeated float data = 9;
}

service MatMul{

	rpc RequestMatMul(MatMulInitRequest) returns (MatMulInitResponse);
	rpc MatrixMultiply(stream MatrixChunk) returns (stream MatrixChunk);
}